fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/AggregatedMaterializedViewRewriter.java

```diff
@@ -476,8 +479,17 @@ protected OptExpression createUnion(OptExpression queryInput, OptExpression view
             logMVRewrite(mvRewriteContext, "Rewrite aggregate with union failed: rewrite aggregate for union failed");
             return null;
         }
-        return createNewAggregate(rewriteContext, queryAgg, newAggregations, aggregateMapping,
+        OptExpression result = createNewAggregate(rewriteContext, queryAgg, newAggregations, aggregateMapping,
                 unionExpr, newProjection);
+
+        if (rewriteContext.getUnionRewriteQueryExtraPredicate() != null) {
+            LogicalFilterOperator filter =
+                    new LogicalFilterOperator(rewriteContext.getUnionRewriteQueryExtraPredicate());
+            result = OptExpression.create(filter, result);
```

Comment:
Operator's `setPredicate` is Deprecated. It may cause column resolve failure if directly used it.
```
    @Deprecated
    public void setPredicate(ScalarOperator predicate) {
        this.predicate = predicate;
    }
```